thunder/torch/__init__.py

```diff
@@ -187,6 +187,68 @@ def is_cuda(a: TensorLike, /) -> bool:
 register_method("size", size)
 
 
+@torchsymbol(torch.Tensor.type, is_method=True)
+def type(a: TensorLike, dtype: None | str | dtypeLike = None, non_blocking: bool = False, /) -> str | TensorLike:
```

Comment:
A first version of this PR might be:

```
@torchsymbol(torch.Tensor.type, is_method=True)
def type(a: TensorLike, dtype: None | str | dtypeLike = None, non_blocking: bool = False, /) -> str | TensorLike:
  <NotImplementedError if dtype is a str>
  <NotImplementedError if dtype is None>
  <NotImplementedError if non_blocking is True>

  return to(a, dtype)
```

But there's already a lot of the needed work to parse the old torch typestrings, so if that's interesting then we can include it. I made some comments for how this PR might refactor that logic to be a little easier to work with